episodes/basic-gis-with-r-sf.Rmd

```diff
@@ -0,0 +1,245 @@
+
+
+```{r}
+ centroids_buffers <- 
+  st_intersection(centroids_old,single_old_buffer) %>%
+   mutate(n=1)
+
+ centroid_by_buffer <- centroids_buffers %>%
+   group_by(ID) %>%
+   summarise(n = sum(n))
+ single_buffer <- single_old_buffer %>%
+   mutate(n_buildings = centroid_by_buffer$n)
+
+  ggplot() + 
+   geom_sf(data = single_buffer, aes(fill=n_buildings)) +
+   scale_fill_viridis_c(alpha = 0.8,
+                        begin = 0.6,
+                        end = 1,
+                        direction = -1,
+                        option = "B")
+```
+
+### Final output:
+
+Let's map this layer over the initial map of individual buildings.
+
+```{r}
+ggplot() + 
+   geom_sf(data = buildings) +
+   geom_sf(data = single_buffer, aes(fill=n_buildings), colour = NA) +
+   scale_fill_viridis_c(alpha = 0.6,
+                        begin = 0.6,
+                        end = 1,
+                        direction = -1,
+                        option = "B") 
+
+```
+ 
+::::::::::::::::::::::::::::::::::::: challenge 
+ 
+## Challenge: Conservation rules have changed. 
+
+The historical threshold now applies to all pre-war buildings, but the distance to these building is reduced to 10m. Can you map the number of all buildings per 10m fused buffer?
+
+
+:::::::::::::::::::::::: solution 
+ 
+```{r}
+ 
+ old <- 1939 
+ distance <- 10
+ #select
+ old_buildings <- buildings %>%
+   filter(start_date <= old)
+ #buffer
+ buffer_old_buildings <- st_buffer(old_buildings, dist = distance)
+  #union
+ single_old_buffer <- st_union(buffer_old_buildings) %>%
+   st_cast(to = "POLYGON") %>%
+   st_as_sf()  
+ 
+ single_old_buffer <- single_old_buffer %>%
+   mutate("ID"=1:nrow(single_old_buffer))  %>%
+   st_transform(single_old_buffer,crs=4326) 
+ #centroids
+ centroids_old <- st_centroid(old_buildings) %>%
+   st_transform(.,crs=4326)  
+  #intersection
+  centroids_buffers <- st_intersection(centroids_old,single_old_buffer) %>%
+   mutate(n=1)
+ centroid_by_buffer <- centroids_buffers %>% 
+   group_by(ID) %>%
+   summarise(
+   n = sum(n)
+   )
+ single_buffer <- single_old_buffer %>% 
+   mutate(n_buildings = centroid_by_buffer$n)
+  ggplot() + 
+   geom_sf(data = buildings) +
+   geom_sf(data = single_buffer, aes(fill=n_buildings), colour = NA) +
+   scale_fill_viridis_c(alpha = 0.6,
+                        begin = 0.6,
+                        end = 1,
+                        direction = -1,
+                        option = "B") 
+   
+```
+::::::::::::::::::::::::
+
+
+:::::::::::::::::::::::::::::::::::::
+
+
+*Problem: there are many pre-war buildings and the buffers are large so the number of old buildings is not very meaningful. Let's compute the density of old buildings per buffer zone.*
+
+## Area
+
+```{r}
+single_buffer$area <- st_area(single_buffer, )  %>% units::set_units(., km^2)
```

Comment:
I get an error here that package `lwgeom` needs to be installed. Let's make sure that this is covered in the setup.